### PR TITLE
KAZOO-5519: make sure schema defaults match their provided type

### PR DESCRIPTION
**Pull request by fenollp**

### CODE REVIEW

File: applications/ecallmgr/src/ecallmgr_config.erl [hunk 1]
```diff
@@ -127,6 +128,32 @@ as_json_value(V, Default) ->
...
+-spec get_jsons(kz_json:path(), Default) -> kz_json:objects() | Default.
+-spec get_jsons(kz_json:path(), Default, kz_json:path()) -> kz_json:objects() | Default.
+get_jsons(Key) ->
+    get_jsons(Key, 'undefined').
+
+get_jsons(Key, Default) ->
+    case get(Key, Default) of
+        Default -> Default;
+        JObjs ->
+            case lists:all(fun kz_json:is_json_object/1, JObjs) of
```
Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_config.erl]: You could put this in a function so `get_jsons/3` could use it too

File: applications/milliwatt/src/milliwatt.hrl [hunk 2]
```diff
@@ -20,13 +20,11 @@
...
-                                        ,<<"tone">>
-                                        ,kz_json:from_list(
-                                           [{<<"caller_id">>,[]}
-                                           ,{<<"number">>,[<<"5555555551">>]}
-                                           ]))
-       ).
+-define(DEFAULT_TONE, kz_json:from_list(
+                        [{<<"caller_id">>,[]}
+                        ,{<<"number">>,[<<"5555555551">>]}
+                        ])).
```
Comment (jamesaimonetti) [inline applications/milliwatt/src/milliwatt.hrl]: The project drops the matching parens/brackets on these multi-line things onto the next line.

File: applications/milliwatt/src/milliwatt_route_req.erl [hunk 3]
```diff
@@ -17,8 +17,7 @@
 -define(ECHO, kz_json:from_list(
                 [{<<"caller_id">>,[]}
                 ,{<<"number">>,[<<"5555555552">>]}
-                ])
-       ).
+                ])).
```
Comment (jamesaimonetti) [inline applications/milliwatt/src/milliwatt_route_req.erl]: This should be reverted, per project style

File: applications/reorder/src/reorder_route_req.erl [hunk 4]
```diff
@@ -36,7 +36,7 @@ maybe_known_number(ControllerQ, JObj) ->
 -spec get_dest_number(kz_json:object()) -> ne_binary().
 get_dest_number(JObj) ->
     {User, _} = kapps_util:get_destination(JObj, ?APP_NAME, <<"inbound_user_field">>),
-    case kapps_config:get_is_true(?CONFIG_CAT, <<"assume_inbound_e164">>) of
+    case true =:= kapps_config:get_is_true(?CONFIG_CAT, <<"assume_inbound_e164">>, false) of
```
Comment (jamesaimonetti) [inline applications/reorder/src/reorder_route_req.erl]: Seems redundant to compare a `boolean()` to `true`

File: applications/stepswitch/src/stepswitch.hrl [hunk 5]
```diff
@@ -26,17 +26,15 @@
 -define(CCV(Key), [<<"Custom-Channel-Vars">>, Key]).
 
 -define(DEFAULT_AMQP_EXCHANGE_OPTIONS
-       ,kz_json:from_list([{<<"passive">>, 'true'}])
-       ).
+       ,kz_json:from_list([{<<"passive">>, 'true'}])).
```
Comment (jamesaimonetti) [inline applications/stepswitch/src/stepswitch.hrl]: Drop matching paren for the `-define` to new line

File: applications/stepswitch/src/stepswitch.hrl [hunk 6]
```diff
@@ -26,17 +26,15 @@
...
 
 -define(RULES_HONOR_DIVERSION
-       ,kapps_config:get_is_true(?SS_CONFIG_CAT, <<"cid_rules_honor_diversions">>, 'false')
-       ).
+       ,kapps_config:get_is_true(?SS_CONFIG_CAT, <<"cid_rules_honor_diversions">>, 'false')).
```
Comment (jamesaimonetti) [inline applications/stepswitch/src/stepswitch.hrl]: Here too

File: applications/stepswitch/src/stepswitch.hrl [hunk 7]
```diff
@@ -26,17 +26,15 @@
...
 
 -define(RESOURCE_TYPES_HANDLED, [<<"audio">>, <<"video">>, <<"sms">>]).
 
 -define(DEFAULT_EMERGENCY_CID_NUMBER,
-        kapps_config:get_non_empty(?SS_CONFIG_CAT, <<"default_emergency_cid_number">>)).
+        kapps_config:get_ne_binary(?SS_CONFIG_CAT, <<"default_emergency_cid_number">>)).
```
Comment (jamesaimonetti) [inline applications/stepswitch/src/stepswitch.hrl]: Et ici

File: core/kazoo_ast/src/kapps_config_usage.erl [hunk 8]
```diff
@@ -194,26 +194,26 @@ key_to_key_path(?BINARY_MATCH(K)) ->
     end.
 
 guess_type('get_list', Default) -> guess_type_by_default(Default);
-guess_type('is_true', _Default) -> <<"boolean">>;
-guess_type('get_is_true', _Default) -> <<"boolean">>;
-guess_type('get_boolean', _Default) -> <<"boolean">>;
+guess_type('is_true', _) -> <<"boolean">>;
```
Comment (jamesaimonetti) [inline core/kazoo_ast/src/kapps_config_usage.erl]: Prefer to keep `_Default` as it helps the reader remember what is in the arg position.

File: core/kazoo_ast/src/kapps_config_usage.erl [hunk 9]
```diff
@@ -262,7 +264,8 @@ guess_properties(Document, SourceModule, Key=?NE_BINARY, Type, Default) ->
     case undefined =:= Description of
         false -> ok;
         true ->
-            io:format("\nYou need to add the key \"~s\" in ~s\n", [DescriptionKey, ?SYSTEM_CONFIG_DESCRIPTIONS]),
+            io:format("\nYou need to add the key \"~s\" in ~s\n"
+                     ,[DescriptionKey, ?SYSTEM_CONFIG_DESCRIPTIONS]),
```
Comment (jamesaimonetti) [inline core/kazoo_ast/src/kapps_config_usage.erl]: Drop matching paren to new line

File: core/kazoo_ast/src/kapps_config_usage.erl (hunk 8 above)
Comment (fenollp) [inline core/kazoo_ast/src/kapps_config_usage.erl]: I don't know... the variable is mentioned in some clauses anyway. The idea was to remove some noise in this very dense area.

**Comment by fenollp** [comment]
Addressed comments :)

**Pull request merged**